tenseal/cpp/tensors/ckkstensor.h

```diff
@@ -63,6 +59,21 @@ class CKKSTensor : public EncryptedTensor<double, shared_ptr<CKKSTensor>>,
     shared_ptr<CKKSTensor> polyval_inplace(
         const vector<double>& coefficients) override;
 
+    shared_ptr<CKKSTensor> dot_inplace(
+        const shared_ptr<CKKSTensor>& to_mul) override;
+    shared_ptr<CKKSTensor> dot_plain_inplace(
+        const PlainTensor<double>& to_mul) override;
+
+    shared_ptr<CKKSTensor> matmul(const shared_ptr<CKKSTensor> other) {
+        return this->copy()->matmul_inplace(other);
+    }
+    shared_ptr<CKKSTensor> matmul_inplace(const shared_ptr<CKKSTensor> other);
```

Comment:
There is something similar in the encrypted_vector interface. Maybe move them to the encrypted_tensor interface?

Reply:
True! This will require removing the n_jobs parameter from matmul, which I can't remember why I chose to put it there... I will open another PR for the API change after this one.